### DISCUSSION TITLE
Unable to Connect Metamask to Verified Smart Contract on Sepolia Network via Etherscan

### DISCUSSION THREAD

**Discussion by tcrin** [Cyfrin/foundry-full-course-cu#3016] [Q&A]
Hi everyone,

I’m trying to interact with a smart contract that has been verified on the Sepolia network, but I’m having trouble connecting my Metamask wallet to it via Etherscan [My contract](https://sepolia.etherscan.io/address/0xEbD7433DF1E0D66513ccAfbB536f0C1674ED4Dfc#readContract). I’ve followed the steps below, but the connection is not working.

### Steps:
1. **Click the "Connect to Web3" button.**  
   ![Screenshot from 2024-12-09 23-08-56](https://github.com/user-attachments/assets/2b812235-bb08-49d6-af17-621504f53397)

2. **An alert pops up, and I click OK.**  
   ![Screenshot from 2024-12-09 23-09-57](https://github.com/user-attachments/assets/f5c6ec64-c85e-4ff7-b806-221d81bf4ac3)

3. **I select Metamask.**  
   ![Screenshot from 2024-12-09 23-10-45](https://github.com/user-attachments/assets/5ad173a6-3336-4063-8a32-9374121ce08c)

4. **Another alert pops up saying "Please connect to your Web3 provider!" and I click OK.**  
   ![image](https://github.com/user-attachments/assets/eb74027f-c0c4-4574-8b8d-ded9eee6842d)

At this point, the transaction signing window does not appear, and I’m unable to connect to my Metamask wallet.

❓Could anyone help me troubleshoot this issue? 
Any suggestions or documentation would be greatly appreciated!

Thanks in advance!

**Comment by EngrPips** [comment]
Hello @tcrin, connecting etherscan to my metamask has never been a problem so much that I needed docs or a guide on connecting. It is fairly simple and straightforward. I will suggest you update your browser if it needs to be updated and also clear you browser cache and then retry.

**Comment by EngrPips** [comment]
Also share the address of the contract you want to interact with so that I can try it on my end too.

**Reply by tcrin** [reply]
Please try my contract:
https://sepolia.etherscan.io/address/0xEbD7433DF1E0D66513ccAfbB536f0C1674ED4Dfc#readContract

**Reply by EngrPips** [reply]
Connecting was as simple as it has always been 

![Screenshot 2024-12-09 at 17 34 21](https://github.com/user-attachments/assets/1def9a08-9bc6-48bd-9d79-5d09c4652b54)

**Reply by tcrin** [reply]
Oh, then maybe my browser has an issue

**Reply by EngrPips** [reply]
Maybe try another browser.

**Reply by tcrin** [reply]
I will try again and get back to you. Thank you for your support!

**Reply by EngrPips** [reply]
Looking forward to hearing from you.

**Reply by mshakeg** [reply]
Hey @EngrPips  the issue I had is that I had multiple browser wallet extensions installed&enabled so I disabled all except for Metamask and it then worked.

**Reply by EngrPips** [reply]
> Hey @EngrPips the issue I had is that I had multiple browser wallet extensions installed&enabled so I disabled all except for Metamask and it then worked.

Interesting. Thanks for this feedback. Who could have thought that is the problem.

**Comment by Oleh325** [comment]
Got the same issue just now. The MetaMask wallet says it's connected to Etherscan, but the website still shows the "Connect wallet" button, and I can't call any functions

**Comment by BhariGowda** [comment]
There are a few common causes for this  try in order:

**1. Multiple wallet extensions conflict (most common fix):**
Disable all browser wallet extensions except MetaMask, then refresh and reconnect.

**2. Wrong network:**
Make sure MetaMask is set to **Sepolia**  not mainnet or another testnet. Etherscan Sepolia only works when your wallet is on Sepolia.

**3. Browser cache:**
Clear browser cache and cookies, then retry.

**4. Try a different browser:**
Chrome works most reliably with MetaMask + Etherscan.

**5. MetaMask permissions:**
Go to MetaMask → Settings → Connected Sites → check if `sepolia.etherscan.io` is listed. If not, manually connect it.

**6. Hard refresh:**
`Ctrl+Shift+R` (Windows) or `Cmd+Shift+R` (Mac) to force reload.

Most people fix it with step 1 or 2.

**Reply by EngrPips** [reply]
Interesting.
